### ISSUE TITLE
Anybody else had issue training PEFT+vLLM?

### ISSUE THREAD

**Issue by August-murr** [huggingface/trl#3013]
### Reproduction

I've been trying to train a PEFT adapter with vLLM, but the training consistently fails—the reward does not increase at all. I noticed that some users have successfully done this in #2818  and #2873 , so I'm wondering if this issue is on my end or if others have faced similar problems.

Kaggle notebook used: [Notebook link](https://www.kaggle.com/code/augustmurr/training-model-with-vllm-peft-for-shorter-length)
When running the same notebook without PEFTConfig, the training works fine, and the reward increases as expected.

without PEFT:
![Image](https://github.com/user-attachments/assets/2ed64569-355d-4aca-b4c9-f7876b6b9258)
with PEFT:
![Image](https://github.com/user-attachments/assets/943ce95a-90e0-4f16-b3e9-9f1db5691061)
the charts are smoothed out. 

### System Info

- Platform: Linux-6.6.56+-x86_64-with-glibc2.35
- Python version: 3.10.12
- TRL version: 0.16.0.dev0
- PyTorch version: 2.5.1+cu121
- CUDA device(s): Tesla T4, Tesla T4
- Transformers version: 4.49.0
- Accelerate version: 1.2.1
- Accelerate config: not found
- Datasets version: 3.2.0
- HF Hub version: 0.28.1
- bitsandbytes version: not installed
- DeepSpeed version: not installed
- Diffusers version: 0.31.0
- Liger-Kernel version: not installed
- LLM-Blender version: not installed
- OpenAI version: 1.57.4
- PEFT version: 0.14.0
- vLLM version: 0.7.3

### Checklist

- [x] I have checked that my issue isn't already filed (see [open issues](https://github.com/huggingface/trl/issues?q=is%3Aissue))
- [x] I have included my system information
- [x] Any code provided is minimal, complete, and reproducible ([more on MREs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any code provided is properly formatted in code blocks, (no screenshot, [more on code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any traceback provided is complete

**Comment by August-murr** [comment]
@qgallouedec I'm gonna have to ask you to reproduce or at least rerun the code you used to train the https://github.com/huggingface/trl/pull/2873#issuecomment-2663793035 so I can calrify wether the problem is on my side and my script or TRL.

**Comment by AndreiCComan** [comment]
@August-murr I had a similar issue in #2856 which has been fixed. Could you try to run the same MRE I posted in #2856 and confirm you are facing the same issue?

**Comment by cuiyuhao1996** [comment]
I ran into the same problem, even with the latest update.

**Comment by cuiyuhao1996** [comment]
Have you solved the problem? :)

**Comment by August-murr** [comment]
@qgallouedec said he was working on it
@qgallouedec any updates?

**Comment by Techie5879** [comment]
Current setup:

vLLM model running on GPU0, and in another notebook, have set GPU 1 to be only visible (for training). This is from - https://github.com/huggingface/trl/blob/main/docs/source/speeding_up_training.md

```
training_args = GRPOConfig(
    output_dir="Llama-3.2-1B-GRPO4",
    logging_steps=1,
    save_steps=500,
    learning_rate=5e-7,
    adam_beta1 = 0.9,
    adam_beta2 = 0.99,
    weight_decay = 0.1,
    warmup_ratio = 0.05,
    max_grad_norm = 0.1,
    max_steps = 10000,
    per_device_train_batch_size=6,
    num_generations=6,
    lr_scheduler_type="cosine",
    push_to_hub=False,
    bf16=True,
    report_to="wandb",
    use_vllm=True,
    max_prompt_length = max_prompt_length,
    max_completion_length = 512,
)
```

```
trainer = GRPOTrainer(
    model=MODEL_ID,
    processing_class=tokenizer,
    reward_funcs=[
        xmlcount_reward_func,
        soft_format_reward_func,
        strict_format_reward_func,
        int_reward_func,
        correctness_reward_func,
    ],
    args=training_args,
    train_dataset=dataset,
    # peft_config=lora_config,
)
```

With PEFT config, training just doesn't seem to work well, without the PEFT config, works much better and rewards are increasing. trl = 0.16.0, peft=0.15.1

**Comment by qgallouedec** [comment]
We usually use a higher learning rate when using peft. Could you try this?

**Comment by Techie5879** [comment]
@qgallouedec I've tried about 2e-5 with llama 3.2 1b. Using lora rank 64

Do you recommend something else/going higher?
